### PR TITLE
Update profile server to respond with a conflict if already profiling

### PR DESCRIPTION
**Pull request by dalehamel**
It is possible that a per-request profile could coincide with a request to
profile in the background via the profile server. This edge case is currently
unhandled, as we don't check the return code of `AppProfiler.start`.

This adds coverage for this edge case, so that it will respond with an HTTP
conflict, as there is already a profile being taken.

### CODE REVIEW

File: test/app_profiler/profile_server_test.rb [hunk 1]
```diff
@@ -116,10 +116,10 @@ class ProfileServerTest < ProfileServerTestCase
           lines = response.lines
           assert(lines.shift.match?(/HTTP.*200/))
           assert_equal("Content-Type: application/json\r\n", lines.shift)
-          assert_equal("Access-Control-Allow-Origin: *\r\n", lines.shift)
           length_line = lines.shift
           assert(length_line =~ (/Content-Length: (.*)/))
           content_length = Regexp.last_match(1).to_i
+          assert_equal("Access-Control-Allow-Origin: *\r\n", lines.shift)
```
Comment (dalehamel) [inline test/app_profiler/profile_server_test.rb:122]: order needed to move since this test is picky about the order it receives the headers in, and I changed it above to keep the linter happy

**Comment by dalehamel** [comment]
AFAIK we have never actually hit this edge case, but since it is actually **possible** to hit it, I figure we should close that gap.

File: lib/app_profiler/server.rb [hunk 2]
```diff
@@ -67,22 +67,26 @@ def handle(request)
...
+              profile = AppProfiler.stop
+              stop_running
+              response.status = HTTP_OK
+              response.set_header("Content-Type", "application/json")
+              profile_hash = profile.to_h
+              profile_hash["start_time_nsecs"] = start_time # NOTE: this is not part of the stackprof profile spec
+              response.write(JSON.dump(profile_hash))
+            else
+              response.status = HTTP_CONFLICT
+              response.write("A profile is already running")
```
[the remark this answers is not in the document]
Comment (dalehamel) [reply lib/app_profiler/server.rb:81]: so return early on any http error, but return at end of method if ok?

[the remark this answers is not in the document]
Comment (bmansoob) [reply lib/app_profiler/server.rb:81]: Yes, and It is a nit. 

Intuitively, when we cannot process the request we usually stop processing right there, so it makes sense to just return.
